### DISCUSSION TITLE
What is the minimum torch and torchaudio version allowed to run this fork? Can CUDA 10.1 users run this fork?

### DISCUSSION THREAD

**Discussion by TechVillain** [voicepaw/so-vits-svc-fork#498] [Q&A] [answered]
**"A recommended version doesn't always mean it's required."**

I am a music instructor and I would love to introduce this lovely AI software to our students to try out.

Here in my school we have several Windows 7 Pro 64-bit computers in our classrooms, running on Nvidia GeForce GTX 660M GPU. According to Nvidia, the highest version of graphic driver we can install is 425.31, and the highest CUDA Toolkit we can install would be 10.1.

According to pytorch dot org, with CUDA version 10.1, the highest torch we can install would be: 
_“torch-1.8.1+cu101-cp39-cp39-win_amd64.whl”._

Here, “cu101” in the file name, is referring to CUDA 10.1. 

Any torch version higher than 1.8.1, will have a higher “cu” number attached in the whl file name, such as: 
_“torch-1.10.0+cu102-cp36-cp36m-win_amd64.whl”_, or
_“torch-1.13.0+cu116-cp310-cp310-win_amd64.whl”_, etc.

**In other words, our school can not install any torch higher than version 1.8.1.**

In the non-fork so-vits-svc-4.0 program folder, there is a file called “requirements.txt”. We opened that file, and can see it says “torch==1.13.1”. Can we assume torch version 1.13.1 is the lowest minimum requirement for so-vits-svc-4.0 fork to run?

**Does it mean we can not install your amazing software on our school’s computers, because our Nvidia GPU are too old, and can’t reach your required pytorch 1.13.1 version? Or maybe it doesn’t matter, a lower pytorch 1.8.1 version can still run?**

Too bad! My colleagues have already trained several G_43200.pth models on their home computers, and they can just simply copy these models to our school’s computers and start the voice inference right away. We don’t need to train on the classroom’s computers, we just need to infer on existing models, to demonstrate to our students. Inferring takes an awful lot less of GPU powers to do.

Has anyone tested this fork on CUDA 10.1?

Please let me know. So, should I give up? Is it a death penalty for our students to see this?

![CUDA   Torch Version](https://user-images.githubusercontent.com/131915615/234696390-246ff5a8-c448-4cd5-96bc-323d724e4e90.PNG)

**Comment by 34j** [comment, marked as answer]
Hello. Since the GTX660M has only 2GB of VRAM, I don't think this program will work anyways. I would recommend using colab. Thanks for your interest.

**Reply by TechVillain** [reply]
Dear 34j, 

To be honest, we already have all the G_ models trained, and we only need to infer small pieces of 10 to 20 seconds of singing vocal, to show our students. 2GB of VRAM is more than enough for light inference jobs like this.

We can't use Colab or other Cloud in the classroom, because we need teach our students how to install svc on a local Windows computer.

### Could you please at least provide some info on if pytorch 1.8.1 with CUDA 10.1 meet the requirements to run?

Because many of our students own GTX 700M, GTX 800M, even GTX 900M series with 6 GB of VRAM or more. However, any GPU below GTX 920M is stuck with Nvidia 425.31 driver, and 10.1 CUDA Toolkit, and therefore can not install pytorch 1.13.1.

### There are so many people who have great VRAMs of GTX, but with a low version of Nvidia and CUDA software, want to use svc. Can you help them?

Please 34j! Thank you so much! Our school have many computers with GTX 920M and below, and we want to run your software. Please help us!

**Reply by 34j** [reply]
I am surprised that you would even ask without trying it yourself. Anyway, the only version supported is CUDA 11.8, and we have absolutely no intention of ever supporting anything less than 11.8. However, CPU inference might be still possible.
